### PR TITLE
Scroll of permanent night

### PR DESCRIPTION
**Pull request by keepermanj**
Create a way for artifact consumables to add permanent effects to the user. Use this to give the Vampire Lord a scroll of permanent night. That artifact scroll permanently adds a source of darkness to the user.

### CODE REVIEW

File: effect.cpp [hunk 1]
```diff
@@ -332,6 +332,20 @@ string Effect::Lasting::getDescription() const {
   return desc.substr(0, desc.size() - 1) + " for some turns.";
 }
 
+void Effect::Permanent::applyToCreature(WCreature c, WCreature attacker) const {
+  c->addPermanentEffect(lastingEffect);
+}
+
+string Effect::Permanent::getName() const {
+  string desc=LastingEffects::getName(lastingEffect);
```
Comment (miki151) [inline effect.cpp]: Spaces around = and + below.

File: item_type.cpp [hunk 2]
```diff
@@ -285,6 +285,10 @@ static int getEffectPrice(Effect type) {
       [&](const Effect::Lasting& e) {
         return LastingEffects::getPrice(e.lastingEffect);
       },
+      [&](const Effect::Permanent& e) {
+        //Permanent effects will probably be from consumable artifacts.
+        return LastingEffects::getPrice(e.lastingEffect)*100;
```
Comment (miki151) [inline item_type.cpp]: Spaces around *

File: lasting_effect.cpp [hunk 3]
```diff
@@ -536,7 +536,7 @@ const char* LastingEffects::getName(LastingEffect type) {
     case LastingEffect::MAGIC_VULNERABILITY: return "magic vulnerability";
     case LastingEffect::MELEE_VULNERABILITY: return "melee vulnerability";
     case LastingEffect::RANGED_VULNERABILITY: return "ranged vulnerability";
-    case LastingEffect::DARKNESS_SOURCE: return "source of darkness";
+    case LastingEffect::DARKNESS_SOURCE: return "night"; //This adjective fits on-screen properly.
```
Comment (miki151) [inline lasting_effect.cpp]: "source of darkness" should fit fine on screen, can you tell me where it doesn't?

**Comment by SoftMonster** [comment]
IZt came out as "Scroll of permanent source of darkness"

**Comment by SoftMonster** [comment]
IZt=it

**Comment by miki151** [comment]
How about "darkness"?

**Comment by SoftMonster** [comment]
Typing on a phone :)

**Comment by SoftMonster** [comment]
Ok

**Comment by keepermanj** [comment]
It now creates a "scroll of eternal darkness", which sounds cool and also fits nicely on my (low res) screen.

**Comment by miki151** [comment]
I would prefer just darkness :)

**Comment by keepermanj** [comment]
The eternal is thrown in to show it is a permanent effect. I can remove it if you like.

**Comment by miki151** [comment]
Yes, please.

**Comment by keepermanj** [comment]
The system is also able to create potions of eternal speed, mushrooms of eternal rage etc.

**Comment by keepermanj** [comment]
Will remove for now.

**Comment by miki151** [comment]
You are right, there should be some distinction. How about permanent? Does
it fit?

**Comment by keepermanj** [comment]
Not on my monitor. I think "endless" fits.

**Comment by keepermanj** [comment]
"permanent darkness scroll" would probably fit but not "permanent scroll of darkness"

**Comment by keepermanj** [comment]
It is "permanent scroll of darkness [1s]"

**Comment by miki151** [comment]
Can you send me a screenshot please?

**Comment by keepermanj** [comment]
OK. One moment. Need to recompile again.

**Comment by SoftMonster** [comment]
Hi Michal,

I pasted the screenshot you wanted below.


[image: Inline image 1]

Thanks for looking at it for me :)

**Comment by miki151** [comment]
I think it got lost :)

Could you send it by regular email? Thanks.

**Comment by keepermanj** [comment]
Sent.

**Comment by keepermanj** [comment]
Should be ready for merging now?

**Pull request merged**